src/collections/Menu/Menu.js

```diff
@@ -128,6 +145,38 @@ class Menu extends Component {
   static Item = MenuItem
   static Menu = MenuMenu
 
+  renderItems() {
+    const { items, onItemClick } = this.props
+
+    this.trySetState({ activeIndex: _.findIndex(items, ['active', false]) || -1 })
+
+    return _.map(items, (index, item) => {
+      const { content, childKey, name, onClick, itemProps } = item
+
+      const active = this.state.activeIndex === index
+      const finalKey = childKey || [content, name].join('-')
+
+      const handleClick = (e, { itemName, itemIndex }) => {
```

Comment:
Prop changes generally result in a component re-rendering. Since this function is being creating new for each item each time `renderItems` is called, these `MenuItem`s will always be receiving new props on each render. This should probably use a similar pattern of having `Menu` define a `handleItemClick` callback like the `Dropdown` does.